scripts/dandi_ephys_uploads.py

```diff
@@ -22,50 +40,105 @@ def get_creds():
+    parser.add_argument('--nwb_folder_path', type=str)
+    parser.add_argument('--dandiset_id', type=str)
     parser.add_argument('--sess_id', type=str)
     parser.add_argument('--project_id', type=str)
     parser.add_argument('--subject_id', type=str)
     args = parser.parse_args()
     set_env()
-    print(validate.validate(args.dandi_file))
+    nwb_folder_path = args.nwb_folder_path
+    dandiset_id = args.dandiset_id
     dandi_set = dandi()
     dandi_set.dandi_authenticate()
-    dandi_dataset = dandi_set.get_dandiset(args.dandi_val)
-
-    status_probes = []
-    path = 'sub_' + args.subject_id + '/' 'sub_' + args.subject_id + 'sess_' + args.sess_id + '/' +  'sub_' + args.subject_id + '+sess_' + args.sess_id + '_ecephys.nwb'
-    status = dandi_dataset.iter_upload_raw_asset(
-        args.dandi_file,
-        asset_metadata={
-            'path': path,
-            "dandiset": str(dandi_dataset)})
-    dir = os.path.dirname(args.dandi_file)
-    probes = ["probeA", 'probeB', 'probeC', 'probeD', 'probeE', 'probeF']
-    for i in glob(join(dir, "*" + 'probe' + "*.nwb")):
-        for probe in probes:
-            if probe in i:
-                print(i, probe)
-                status_probes.append(
-                    dandi_dataset.iter_upload_raw_asset(
-                        i,
-                        asset_metadata={
-                            'path': 'sub_' + args.subject_id + '/' 'sub_' + args.subject_id + 'sess_' + args.sess_id + '/' +  'sub_' + args.subject_id + '+sess_' + args.sess_id + '+' + probe + '_ecephys.nwb',
-                            "dandiset": str(dandi_dataset)}))
-    print(list(status))
-    for i in status_probes:
-        print(list(i))
-
-    fb.start(fb.get_creds())
+    dandi_dataset = dandi_set.get_dandiset(dandiset_id)
+    session_id = args.sess_id
+    project_id = args.project_id
+    subject_id = args.subject_id
+  
+
+    set_env()
+    dandiset_path = nwb_folder_path
+    assert os.getenv("DANDI_API_KEY"), (
+        "Unable to find environment variable 'DANDI_API_KEY'. "
+        "Please retrieve your token from DANDI and set this environment variable."
+    )
+
+    # Create the session directory
+    directory_path = os.path.join(nwb_folder_path)
+    dandi_path = os.path.abspath(directory_path)
+    dandi_path_set = directory_path  + "/" + dandiset_id
+    print("PATHS: ", directory_path, dandi_path, dandi_path_set)
+    if not os.path.exists(directory_path):
+        os.makedirs(directory_path)
+        os.makedirs(dandi_path_set)
+
+
+    # Rename the nwb file to include the session id
+    src = os.path.join(nwb_folder_path,  "spike_times.nwb")
+    dst = os.path.join(directory_path, session_id + ".nwb")
+
+    # Move the nwb file to the session directory
+    shutil.move(src, dst)
+    
+
+    dandi_download(urls=r'https://dandiarchive.org/dandiset/' + dandiset_id + '/draft/', output_dir=str(dandi_path), get_metadata=True, get_assets=False)
```

Comment:
So this download the entire dandiest, correct?

Reply:
Will this force each upload of one session trigger the download of the entire dataset?

Reply:
This only downloads the metadata asssociated with the set, not the assets. So just the dandiset.yaml